### ISSUE TITLE
Nyan Cat trademark?

### ISSUE THREAD

**Issue by stas730** [minetest/minetest_game#1647]
Nyan Cat is used in Minetest but original Nyan Cat is non-free.
Free, fair or illegal use?

**Comment by sfan5** [comment]
[mods/nyancat/license.txt](https://github.com/minetest/minetest_game/blob/master/mods/nyancat/license.txt#L18)

**Comment by red-001** [comment]
https://github.com/minetest/minetest_game/blob/master/mods/nyancat/license.txt#L21

**Comment by dsohler** [comment]
Not sure if CC BY-SA 3.0 applies.

http://www.bbc.com/news/technology-22397446  
http://www.theverge.com/2013/5/2/4293484/nyan-cat-keyboard-cat-creators-sue-scribblenauts-over-easter-eggs

**Comment by stas730** [comment]
I know.
Original Nyan Cat video is non-free.

**Comment by paramat** [comment]
From theverge link:
"Torres stresses the distinction between commercial and noncommercial use: "I have no issues with Nyan Cat being enjoyed by millions of fans as a meme, and I have never tried to prevent people from making creative uses of it that contribute artistically and are not for profit.""

**Comment by dsohler** [comment]
I wasn't able to find an official statement, that Nyan Cat is CC BY-SA 3.0.

**Comment by red-001** [comment]
Copyright isn't really the issue here, the main issue seems to be trademark related.

**Comment by sfan5** [comment]
@dsohler nyan cat is not cc by-sa, our texture is.

**Comment by dsohler** [comment]
That wasn’t the question @stas730 had, as far as I understand the issue.

Which is actually a real issue and shouldn’t be closed with linking to CC BY-SA 3.0 for the textures.

Using non-free (trademarked) Nyan Cat (fair use? trademark-owner-does-not-care-because-not-commercial use?) IS something that should be clarified and clearly stated.

**Comment by sofar** [comment]
I'm for removing the `nyan` mod entirely.

**Comment by dsohler** [comment]
Maybe not removing it, but making it 100 percent clear what the license of this trademark is and that it is not allowed to use it in a commercial context without permission by the trademark owners.

Simply closing the issue and pointing out that the textures that are used in Minetest are CC BY-SA 3.0 is not a valid solution for this issue.

**Comment by Fixer-007** [comment]
Rename Nyan Cat to Mine Cat? Texture is made from scratch, right?

**Comment by paramat** [comment]
Well, it's obviously a Nyan Cat, using a different name would look like an insulting attempt to avoid copyright conflicts and would irritate the creator even more.
I agree with dsohler, at the least a clear statement is needed in the mod that it is not for commercial use and will be removed on request. It is a part of Minetest's character so i would prefer to keep it.

**Comment by rubenwardy** [comment]
You just need to change enough it so it no longer infringes trademarks

**Comment by sofar** [comment]
Or just delete it. Much easier.

**Comment by VanessaE** [comment]
The nyancat image I created was drawn by hand, with only basic inspiration from the original, all fresh colors, and a clearly different style for the face.  Because it does not purport to be the original work, and does not attempt to replace or masquerade as the original in the contexts and markets where the original is used (e.g. swag of some kind), it does not infringe on the original trademark.  Besides, it's an entirely original work that only looks similar to nyan cat.

**Comment by sofar** [comment]
@VanessaE that's an excellent testimony, but, even things that look extremely similar are a potential trademark violation.

**Comment by VanessaE** [comment]
That would only be true if Minetest's imagery were used in the same market as the original.

See the various Apple Corps (the record co. owned by The Beatles) vs. Apple Computer fights (usually Apple Corps won).  In particular, the 1978 case laid the foundation for what I wrote above.

**Comment by paramat** [comment]
I don't think it needs changing, it's obviously a Nyan Cat and we want it to look like one, the texture is much loved too.
Seeing the creator's opinion about non-commercial usage i think we wold be ok, but a statement that it will be removed on request might be a good idea.
I feel it's best our texture states that it is inspired by, or, derived from, the original Nyan Cat, instead of trying to pretend there's no trademark issue (which would likely irritate the creator much more).

**Comment by sofar** [comment]
IANAL, but I like to err on the side of caution. And the side of caution would remove any and all reference to "nyan cat" which is a registered trademark. Even if the image is different, we're still liberally using the words in that order.

nuke it from orbit. Just like mese.

**Comment by VanessaE** [comment]
The wording doesn't matter either, because officially the name of the creature is "Pop Tart Cat".  The phrase "Nyan Cat" only came about due to a meme, and was not Torres' idea.

> Christopher Torres said: "Originally, its name was Pop Tart Cat, and I will continue to call it so, but the Internet has reached a decision to name it Nyan Cat, and I’m happy with that choice, too.."

**Comment by VanessaE** [comment]
Correction, he does call it "Nyan Cat" in his trademark application.  Meh.

**Comment by sofar** [comment]
what matter is the USPTO registration that is active for the words "nyan cat"

**Comment by VanessaE** [comment]
In any case, I am strictly opposed to removing nyans or mese from the game, they're just too unique to MT and too widely-used (or just sought after).

If either should come to pass, they'll find their way into gloopblocks.

**Comment by sofar** [comment]
No, minetest's "uniqueness" is absolutely not determined by an easter egg.

**Comment by sorcerykid** [comment]
Why is this issue immediately closed before a thorough vetting process by the community? Am I to understand that sfan5 is the exclusive legal arbiter now for Minetest?

Second of all, whether the image is hand-drawn has absolutely no bearing on legal culpability, since the peculiar likeness constitutes a derivative work. And adaptation is still an exclusive right that can only be exercised by the copyright owner.

Third of all, courts will typically weigh the potential for consumer confusion in making a determination of trademark infringement. Thus, the fact this so-called "hand drawn" rainbow cat is specifically called a Nyan Cat and considered so "unique to Minetest" is cut-and-dry admission of wrongdoing. Moreover, the hand drawn nature is evidence of plagiarism since the actual source of the original image is never prominently disclosed to the end-user of the software, which is just another strike against its unauthorized use.

Fourth of all, trademarks classes come into play only when the mark itself has not gained sufficiently widespread recognition and notoriety that a consumer within the broader marketplace could readily distinguish the source of goods. So the argument about Apple Computer Inc. is not pertinent to this discussion.

Fifth of all, non-profit use should not be conflated with fair use. The later refers to an exemption in U.S. copyright law for purposes of reporting, criticism, analysis, etc. The display, reproduction, public performance, etc. rights of a copyrighted work outside of fair use still requires authorization by the copyright owner even in non-commercial settings unless a license is granted by the copyright owner or by a blanket licensing agency (which is common for musical works, sound recordings, motion pictures, etc.).

Lastly, being "strictly opposed to removing" something just because its "widely used" is not a sufficient justification for violating copyright and trademark law nor is it setting a good example for the open-source community.

_Disclaimer: I am not an attorney. The information hereinbefore does not constitute legal advice nor consultation. I am merely sharing my personal knowledge of intellectual property rights. I would recommend referring this issue to qualified legal counsel._

**Comment by paramat** [comment]
Perhaps then we could remove the mod from MTG and have it around as an optional mod which most people will add themselves to not break their worlds?

**Comment by C1ffisme** [comment]
@paramat That sounds agreeable. What do we alias all nyan cat nodes into, though?

**Comment by paramat** [comment]
They wouldn't be aliased in MTG, if they were, those who want to keep their nyan cats and use the mod would lose their nyan cats. Those who choose to not use the mod can use a mod to alias, or just remove the unknown nodes in their world as they find them.

Best to assume the creator will discover MT's use of the nyan cat, and prepare for that now. The current approach of trying to pretend there is no issue because the texture is freely redrawn, when it is so obviously still a nyan cat, will i expect irritate the creator. If we keep it in MTG it should be credited as being derived from or inspired by the original.

**Comment by sfan5** [comment]
> Why is this issue immediately closed before a thorough vetting process by the community? Am I to understand that sfan5 is the exclusive legal arbiter now for Minetest?

Umm no? I closed this issue because I believed to have resolved it with my reference to the license file.

**Comment by red-001** [comment]
The title should be changed since the issue isn't with the license under which the texture is but the potential  trademark infringement.

**Comment by VanessaE** [comment]
Just as my commentary carries no legal weight, the same is true for yours, @sorcerykid.  "Disclaimer: I am not an attorney" tells us everything we need to know about your comment.

The original Pop Tart Cat author, *who took the "Nyan Cat" name from the "meme community"* explicitly said he is fine with non-commercial usage, which clearly is the case for MT (at least as distributed from its official sources).

As I said, one of the primary tests for trademark infringement is whether you're a name similar to the trademarked product in the same context and market as the trademarked product.  Minetest does not.

Another is whether the use of the trademark risks harm to the name or market standing of the trademarked product.  Minetest does not.

Another test of trademark validity is the simplicity of the trademark - unless something's changed in the past few years, excessively-simple designs/marks are excluded from trademark.  The trademark on the words "Nyan cat" explicitly state in the trademark application that it's literally just those two words, in plain font, with nothing fancy.  How the hell that got granted, I'll never know.

Lastly, I engaged in an email discussion with Torres back in 2012 about this. My initial email was to ask for licensing terms.  I don't need to share the discussion, but needless to say, he tried to make legal threats with no basis, and the discussion went nowhere.  In the last email I received from him, he wrote:

> I think we (and my copyright attorney) can all agree that this does NOT
> fall under fair use, since this game is using my exact artwork extremely blatantly. 

This is what prompted me to re-draw the image so as to comply with his obvious requirement.  I was not contacted again by any party associated with or working for Torres in any capacity (that I'm aware of).

So to recap:  it's my image, I drew it from scratch in response to the author's complaint, and made it only look something like the original, and it's used in a non-commercial setting (which Torres explicitly stated elsewhere is okay).  

There is no trademark or copyright violation here.

**Comment by red-001** [comment]
Torres might be making baseless treats but I would like to point out that he managed to force Warner Brothers to settle.

**Comment by VanessaE** [comment]
I wouldn't say he forced anything.  He and the guy who came up with "keyboard cat" sued over commercial use of their likenesses.  WB settled because they were clearly in the wrong here and it wasn't worth their time to argue.

**Comment by sorcerykid** [comment]
Vanessa E, my disclaimer does not discount the quality of the information I provided. In the United States it is due diligence for _anyone_ publishing legal information to include such a disclaimer (apparently you have never heard of "unauthorized practice of law", which carries criminal penalties). I am well enough versed in reviewing intellectual property rights, as that was my former occupation.

Furthermore, let's be clear on one point: The burden of proof is entirely your court since the potential liability is a result of your tortuous actions not mine.

> As I said, one of the primary tests for trademark infringement is whether you're a name
> similar to the trademarked product in the same context and market as the trademarked
> product.  Minetest does not.

The _primary_ test is the likelihood of consumer confusion. And the gauge for "confusingly similar" usage, varies depending on the distinctiveness of the mark. Even if the trademark is registered in only one class,that  still doesn't preclude the trademark owner from litigating for unauthorized use in another class, if the usage is likely to cause any degree of consumer confusion.

> Another is whether the use of the trademark risks harm to the name or market standing of
> the trademarked product.  Minetest does not.

That is a subject of trademark dilution, which I think is beyond the scope of this discussion. Let's tackle one issue at a time.

> Another test of trademark validity is the simplicity of the trademark - unless something's changed in the past few years, excessively-simple designs/marks are excluded from trademark.

Simplicity with respect to trademark law means that a brand must be distinctive. A business cannot trademark commonplace or descriptive words and phrases like "Cheeseburger" or "Hometown Grocery Store". A business also cannot trademark basic, everyday symbols and designs, like a circle with a solid black border or a red diagonal line.

> The trademark on the words "Nyan cat" explicitly state in the trademark application that it's literally just those two words, in plain font, with nothing fancy. How the hell that got granted, I'll never know.

The fact you cannot understand how the trademark was approved speaks volumes about your knowledge of the subject. I paid for a registered U.S. trademark for my business (which is one word in a plain font with nothing fancy), and I am very familiar with the USPTO application guidelines.

> Lastly, I engaged in an email discussion with Torres back in 2012 about this. My initial
email was to ask for licensing terms.  I don't need to share the discussion, but needless
to say, he tried to make legal threats with no basis, and the discussion went nowhere.  In
the last email I received from him, he wrote:

Your lack of transparency is concerning. If you refuse to disclose all pertinent communication with the trademark owner, then I think it is only fair that the core developers err on the side of caution and remove the Nyan Cat accordingly.

> This is what prompted me to re-draw the image so as to comply with his obvious requirement.  I was not contacted again by any party associated with or working for Torres in any capacity (that I'm aware of).

Not being contacted by a trademark owner is not a proof of compliance.

> So to recap:  it's my image, I drew it from scratch in response to the author's complaint,
and made it only look something like the original, and it's used in a non-commercial
setting (which Torres explicitly stated elsewhere is okay).

Frankly, I don't care. It's not even my intellectual property. But you'd be wise to step back and consider whether your personal  take priority over the best interests of the Minetest community at large.
Vanessa E, my disclaimer does not discount the quality of the information I provided. In the United States it is due diligence for _anyone_ publishing legal information to include such a disclaimer (apparently you have never heard of "unauthorized practice of law", which carries criminal penalties). I am well enough versed in reviewing intellectual property rights, as that was my former occupation.

Furthermore, let's be clear on one point: The burden of proof is entirely your court since the potential liability is a result of your tortuous actions not mine.

> As I said, one of the primary tests for trademark infringement is whether you're a name similar to the trademarked product in the same context and market as the trademarked product.  Minetest does not.

The _primary_ test is the likelihood of consumer confusion. And the gauge for "confusingly similar" usage, varies depending on the distinctiveness of the mark. Even if the trademark is registered in only one class,that  still doesn't preclude the trademark owner from litigating for unauthorized use in another class, if the usage is likely to cause any degree of consumer confusion.

> Another is whether the use of the trademark risks harm to the name or market standing of the trademarked product.  Minetest does not.

That is a subject of trademark dilution, which I think is beyond the scope of this discussion. Let's tackle one issue at a time.

> Another test of trademark validity is the simplicity of the trademark - unless something's changed in the past few years, excessively-simple designs/marks are excluded from trademark.

Simplicity with respect to trademark law means that a brand must be distinctive. A business cannot trademark commonplace or descriptive words and phrases like "Cheeseburger" or "Hometown Grocery Store". A business also cannot trademark basic, everyday symbols and designs, like a circle with a solid black border or a red diagonal line.

> The trademark on the words "Nyan cat" explicitly state in the trademark
> application that it's literally just those two words, in plain font, with nothing fancy.
> How the hell that got granted, I'll never know.

The fact you cannot understand how the trademark was approved speaks volumes about your knowledge of the subject. I paid for a registered U.S. trademark for my business, and I am very familiar with the USPTO application guidelines.

> Lastly, I engaged in an email discussion with Torres back in 2012 about this. My initial
email was to ask for licensing terms.  I don't need to share the discussion, but needless
to say, he tried to make legal threats with no basis, and the discussion went nowhere.  In
the last email I received from him, he wrote:

Your lack of transparency is concerning. If you refuse to disclose all pertinent communication with the trademark owner, then I think it is only fair that the core developers err on the side of caution and remove the Nyan Cat accordingly.

> This is what prompted me to re-draw the image so as to comply with his obvious
> requirement.  I was not contacted again by any party associated with or working for Torres
> in any capacity (that I'm aware of).

Not being contacted by a trademark owner is not a proof of compliance.

> So to recap:  it's my image, I drew it from scratch in response to the author's complaint,
and made it only look something like the original, and it's used in a non-commercial
setting (which Torres explicitly stated elsewhere is okay).

I think you'd be wise to step back and consider whether your personal attachment to the hand-drawn Nyan Cat takes precedence over the best interests of the Minetest community at large.

**Comment by VanessaE** [comment]
> Your lack of transparency is concerning

My "lack of transparency" is because it was several emails that are entirely irrelevant since Torres has already made it clear that he didn't want his actual image used, and has since said that non-commercial usage is fine.  Minetest is non-commercial.

> I think you'd be wise to step back [...]

My "step-back" is as follows:  If the Minetest devs decide that want to delete the images, let them.  I can't and won't attempt to stop that action, but you can be sure I'll use my images in my mod, under the "non-commercial is okay" condition Torres has already made clear.

**Comment by rubenwardy** [comment]
> <sorcerykid> Your lack of transparency is concerning. If you refuse to disclose all pertinent communication with the trademark owner, then I think it is only fair that the core developers err on the side of caution and remove the Nyan Cat accordingly.

VanessaE is a trusted member of the community, potentially more so than most core devs (as she's been around for longer). Also, how do you know that these emails haven't been shared with celeron55 / core devs already?

> <VannesaE> The trademark on the words "Nyan cat" explicitly state in the trademark application that it's literally just those two words, in plain font, with nothing fancy. How the hell that got granted, I'll never know.

This is just obviously wrong, sorry :/  Trademarks can cover just words, they don't have to cover logos as well. Nyan Cat is distinctive as a combination, imo - Nyan is a made up word based on a sound (afaik).

> <VanessaE> As I said, one of the primary tests for trademark infringement is whether you're a name similar to the trademarked product in the same context and market as the trademarked product. Minetest does not.
>
>Another is whether the use of the trademark risks harm to the name or market standing of the trademarked product. Minetest does not.

Whilst it is true that our use of the trademark doesn't currently cause confusing, that's not how the law works. **If a company doesn't enforce a trademark they lose it**. This is why you get bogus trademark claims.

> said that non-commercial usage is fine. Minetest is non-commercial.

This means the license should probably be NC then :/

-----------

Whilst I like this easter egg, I feel like it's probably not worth any risk. We can add other easter eggs in its place

**Comment by dsohler** [comment]
@VanessaE If I re-draw a texture resembling Mickey Mouse and apply it and call the thing “Mickey Mouse” it still is a potentially illegal use of the Mickey Mouse trademark owned by the Walt Disney Company, even if I own the copyright for the texture I created.

There is absolutely no difference with Nyan Cat. You created the textures resembling Nyan Cat, the texture gets applied and the result is called “Nyan Cat”. This means that you own the copyright for the texture. But releasing it unter CC BY-SA 3.0 does not make using the Nyan Cat trademark legal.

Oh, and according to your description of the mail communication the trademark owners clearly said that this (I guess “this” means using Nyan Cat in Minetest) does not fall under fair use … And it is still used? Wow. Ballsy devs :smile:

**Comment by Thomas--S** [comment]
And how about simply changing the NyanCat to a MooGNU as it was suggested earlier in this topic? I think this would maybe be the the easiest solution because

- backwards compatibility (we only need to register an alias)
- There is no big change for players
- Probably only a minimal legal risk (_Disclaimer:_ I'm not a lawyer!)

**Comment by sofar** [comment]
> Minetest is non-commercial.

This is clearly nonsense and incorrect:

Nothing in the GPL forbids commercial companies from using GPL software.

Therefore, the GPL is clearly agnostic about "commercial" activities. It only prevents companies from charging for GPL software, but companies can both *use* and *distribute* GPL software.

Second, the license of the texture is CC-BY-SA. Now in this discussion, it's entirely clear that the trademark holder is against any commercial use.

This changes the license of the texture made by VanessaE to CC-BY-SA-NC.

This prohibits any company from *USING* minetest_game or from *RUNNING* a minetest_game based server. -NC is very ambiguously defined and not even a company offering Tutoring services or Summer camps could use it.

This is in clear conflict with the spirit of `minetest_game`. It has always meant to be permissively licensed, and we have never accepted a -NC licensed component.

So, I do not agree with labeling minetest -NC and I strongly object against anyone attempting to change the license which would be a breach of trust to developers.

Sorry, I'm not giving up about this. It sucks, but, the cat has to go.

Someone freehand paint a cute and fuzzy animal in gimp and I'll make a PR that replaces it.

**Comment by VanessaE** [comment]
I was considering making a PR to replace the nyan blocks with entirely new objects.

Since it would need to be kind of cutesy/silly, I was thinking of a block depicting a cartoonish dog (at least in part), with a body composed of open-faced peanut butter and jelly sandwiches.  The front and back would have purple blotches representing grape jelly, the top/bottom/sides would be the peanut butter.  The rainbow block would perhaps be replaced with a white block with a scattering of colorful candies.  Overall the layout would be about the same as the existing work, but the resultant images would constitute an explicit and deliberate parody of Nyan Cat.

Its official name would be "PB&J Pup".

A google search shows that that term (with or without the "&") appears to be unclaimed (and the few results I did find were just the usual search engine gibberish, like two hash tags close together, or license plate suggestions).  Therefore, if someone thinks this is a good idea, I'll try to draw appropriate images.  To that end, I lay claim to the terms "PB&J Pup" and "pbj_pup" and the images I described above, as of today's date.

Let's see the copyright or trademark trolls attack THAT.

**Comment by sofar** [comment]
Parodies are an explicit exemption to trademark law, so, I applaud and welcome to PB&J Pup!

**Comment by VanessaE** [comment]
Then I will be submitting a pull request shortly.

**Comment by dsohler** [comment]
> PB&J Pup

Since Minetest is an international game and this whole "PB&J" stuff is rather uncommon outside the US (plus the name does not have a good flow - the name is just clumsy) I'm not quite sure about this.

> Let's see the copyright or trademark trolls attack THAT.

The [MooGNU](https://archive.org/details/M00GNU) has peanut butter toast sides :smile:

(Plus: It is somewhat common in the FLOSS community and an already established thing with a good flowing name and it is compatible to the minetest_game license because it is CC BY-SA 3.0).

**Comment by sofar** [comment]
it's not GNU/Minetest. Definitely the GNU needs to run away before I make him into (delicious) steak.

**Comment by VanessaE** [comment]
Actually rather than a pull request, I'll just post the images right here, and you guys can handle integrating them into a new mod to replace the offending cat:

![pbj_pup_back](https://cloud.githubusercontent.com/assets/1733537/24090645/8099fdac-0d18-11e7-9ce1-7e6abaacee86.png)  ![pbj_pup_candies](https://cloud.githubusercontent.com/assets/1733537/24090646/80a38bec-0d18-11e7-9c0c-b1ab1de32506.png)  ![pbj_pup_front](https://cloud.githubusercontent.com/assets/1733537/24090647/80a44532-0d18-11e7-89a1-b58a9ede8fa2.png)  ![pbj_pup_jelly](https://cloud.githubusercontent.com/assets/1733537/24090649/80a82f8a-0d18-11e7-8d1f-ec4252561cb3.png)  ![pbj_pup_sides](https://cloud.githubusercontent.com/assets/1733537/24090648/80a7f2b8-0d18-11e7-8421-a9d498134570.png)

Here's a zip of a very minimal mod using the above images.  It includes the gimp XCFs (see pbj_pup/xcf) but does not include any of the spawning code and such - just a couple of node defs based on the current nyan cat code:

[pbj_pup-20170320-0254.zip](https://github.com/minetest/minetest_game/files/854057/pbj_pup-20170320-0254.zip)

**Comment by sofar** [comment]
I'll be most happy to. May I suggest the following attribution/copyright/licensing?

(C) 2017 VannessaE
  - Code: LGPL-2.1+
  - Images: CC-BY-4.0

**Comment by VanessaE** [comment]
Agreed.

**Comment by VanessaE** [comment]
Except spell my name right :smile:

**Comment by sofar** [comment]
Give me some time to soup it up a bit :)

**Comment by VanessaE** [comment]
Feel free to use as much bouillon as you need. :)

**Comment by dsohler** [comment]
Will you draw a the pup from top and bottom, too? Or is the poor little thing surrounded by PBJ sandwiches? :smile:

**Comment by paramat** [comment]
![tac_nayn](https://cloud.githubusercontent.com/assets/3686677/24103837/74879ea6-0d78-11e7-8653-2502fc9f6075.png)

![tac_nayn_rainbow](https://cloud.githubusercontent.com/assets/3686677/24103843/77263d8e-0d78-11e7-9ae6-f9a2bb6d433e.png)

If you're going to parody, this is cuter (cats are cuter than dogs) and still a cat, and already a mod, not sure if too close to being a nyan cat or not.
http://wiki.minetest.net/Mods/Tac_Nayn

**Comment by paramat** [comment]
I think there should be a way to keep the nyan cat by using the nyancat mod as an optional mod. Many will certainly want to keep it in their worlds (like me) so this should be possible somehow. I guess we would just remove the alias to the PB&J Pup.

**Comment by rubenwardy** [comment]
Nyancat should be moved out of MTG, and any replacements should be made as new mods in MTG, imo. Not sure about backwards compat / aliases / lbms

**Comment by paramat** [comment]
I just realised a nyan cat mod separate from MTG could not be accepted as a mod because it would have to be NC.
So unfortunately, as much as i love the nyan cat, i think it has to be replaced. This will not be popular :]

**Comment by DS-Minetest** [comment]
Couldn't we also leave the nyan cat as it is and make a mod that replaces it for people using minetest for commercial things?

**Comment by rubenwardy** [comment]
The forum does not disallow NC mods, only ND mods

**Comment by DS-Minetest** [comment]
The forum disallows nyan dog mods?!

**Comment by dsohler** [comment]
Yes, best would be extracting Nyan Cat for minetest_game completely without aliasing it to the replacement and offer it as separate C BY-NC mod via forum.

**Comment by rubenwardy** [comment]
NC and nd = Non commercial and no-derivatives

I don't really see the need of a meme like this in minetest game. I feel like it would be better to expand on the mese/alien aspect to add rare blocks

**Comment by paramat** [comment]
> The forum does not disallow NC mods, only ND mods

Ok.

I prefer that the mod is removed from MTG and is available for use as an optional added mod for anyone who wants to keep the nyan cats in their world (many people will want to, it's much loved and is used in constructions). I request there is a way for these people to avoid the nyan cat nodes being aliased into any replacement that is added such as the PB&J Pup or Tac Nayn.

Then otherwise, nyan cat nodes get aliased into the replacement character.

**Comment by VanessaE** [comment]
Disregarding the obvious bias I would have toward my own images :smile:, imho Tac Nayn is not an appropriate replacement.  Frankly, it's ugly and is apparently evil ("Destroyer of worlds"), and besides which, the author thereof surely can claim his own copyright over it.

**Comment by paramat** [comment]
The author is one of our modders, GreenDimond, and it's a cute evil black cat.

**Comment by stujones11** [comment]
I am very sad that Nyan cat has to go but I guess there is no option :'(

I personally favour the idea of using MooGNU as a replacement. at least we'd get to keep the rainbows.

**Comment by tenplus1** [comment]
All of this over a 16x16 representation of a cartoon cat ?!?! what has this world come to...

+1 fo MooGNU

**Comment by stujones11** [comment]
> what has this world come to...

Exactly, that is why I think using MooGNU is kinda apt, no disrespect to the pup, it is very cute.

**Comment by tenplus1** [comment]
You could even cutsey up the Gnu with a  :3  face.

**Comment by tenplus1** [comment]
![alt text](http://i66.tinypic.com/2mw76zr.png "MooGNU")

**Comment by VanessaE** [comment]
@paramat actually I meant the original author of the Tac Nayn character, not the imagery on the Minetest node.

**Comment by sorcerykid** [comment]
I suggest keeping the rainbow textured nodes. Those clearly are not protected by copyright (esp. once they are disassociated from the Nyan Cat) and to remove or alter them would dramatically affect existing builds :(

As for the Nyan Cat substitution, it doesn't matter much to me what character is chosen as long as it fits with the Minetest media license. I mean, if it's half-way decent artistically, I doubt most players will care. I just hope the name is something that is compelling and easy to recall, nothing too esoteric.

**Comment by sorcerykid** [comment]
PS. Since VanessaE is the one that redrew the Nyan Cat perhaps she could be the one to render whatever character is decided upon since her work on the Nyan Cat was first rate. And I like seeing that kind of quality in Minetest.

**Comment by Thomas--S** [comment]
I really like @tenplus1's suggestion. I would suggest to merge his suggestion into master as soon as possible to avoid legal issues. The artistic style can be improved later, if there is the need for this.

**Comment by stujones11** [comment]
> I really like @tenplus1's suggestion.

It's not a bad first effort but I think it still needs some work, particularly the tail-end. Not that I could do any better.

**Comment by paramat** [comment]
I see, i did not know Tac Nayn was not created by Green Dimond.
The GNU is very rough and not cute.
PB&J Pup is ok with me.

**Comment by tenplus1** [comment]
![moognu_back](https://cloud.githubusercontent.com/assets/8145060/24167537/fedb4d22-0e6e-11e7-99b7-8978705bfc2e.png)
![moognu_front](https://cloud.githubusercontent.com/assets/8145060/24167536/fedae7ce-0e6e-11e7-916a-b41c3b34efa5.png)
![moognu_side](https://cloud.githubusercontent.com/assets/8145060/24167538/fedd6d50-0e6e-11e7-9871-e379b6407bd9.png)

**Comment by paramat** [comment]
Sorry i should have realised these are texture examples that are not finished.

**Comment by sofar** [comment]
The moognu leaves a trail of moopoo.

**Comment by sofar** [comment]
seriously though, Minetest is more MIT/copyleft and the GNU gnu does not belong.

**Comment by Thomas--S** [comment]
As far as I know, [minetest](https://github.com/minetest/minetest) is licensed under the GNU Lesser General Public License. So, actually, I think the MooGNU would fit quite well. It would also allow us to keep the rainbows unchanged, which is good in terms of backwards compatibility. I think it may often be a problem for existing buildings, if the former rainbows will be changed to the candy-like blocks of the PB&J Pup.

**Comment by paramat** [comment]
If you want to keep your rainbows, use the settings to keep your nyan cats, and use the separate nyan cat mod.

**Comment by stujones11** [comment]
Regardless of what is used for the 'head' of this thing, the rainbows should stay. Changing those almost forces server owners to install the 'non-commercial-use' nyan cat mod, or at least make their own.

**Comment by paramat** [comment]
Those who use the rainbows in builds usually also use the cat itself.

**Comment by Thomas--S** [comment]
I think the change `cat -> pup` is affecting buildings less than the change `rainbow -> candy`. Cats and pups both belong to the group "animals", but I actually see no connection between a rainbow and candies.

**Comment by stujones11** [comment]
Rainbows are way more prevalent in builds that the cats themselves, in fact in my own experience and from screenshots I've seen the cats are seldom used.

**Comment by dsohler** [comment]
It’s called “a change”. Things change every now and then. It’s a part of the human nature to be able to adapt to changes. We’re all humans … so … :smile:

**Comment by sofar** [comment]
I'm purposely making the new mod not something that you'd build with (hence some of the code) but more of a "special" item. I want to avoid people from building with these blocks entirely. They're out of style anyway. Let them be a fun thing for kids to find that are playing singleplayer. I honestly don't know a MP server admin that can't solve the change.

**Comment by VanessaE** [comment]
If players want a "rainbow" block, but want to avoid the nyan mod, they can use the one in gloopblocks.

**Comment by tenplus1** [comment]
The rainbow block has no issues towards it and should remain in the game since players use these for builds and other mod usage, changing it altogether or requiring a mod to replace it is a no-no.

**Comment by paramat** [comment]
We could add it as a new node, but it can't keep its current name.

**Comment by sofar** [comment]
again, you're asking us to break something irreparably for some user.

**Comment by VanessaE** [comment]
Since it's a good idea for most servers to have gloopblocks installed, how about I add the old nyan rainbow to that mod (in addition to the rainbow it already has), and alias it to the old name?

**Comment by paramat** [comment]
What if people want to use gloopblocks mod along with keeping the nyancat mod? Then you have a conflict of aliases. We could add a rainbow to game possibly but no aliases, too messy combined with what we're doing here.

**Comment by VanessaE** [comment]
in such a case, I'd simply have gloopblocks opt-depend on the nyan mod, detect its presence, and switch-off its alias.

**Comment by VanessaE** [comment]
Since the crux of the problem is the existence of the nyan head and not so much the rainbow, why not do both?  A generic rainbow in one mod, with alias, and the head in another?

**Comment by GreenXenith** [comment]
> it's ugly and is apparently evil ("Destroyer of worlds")

Hey, just flip that frown upside-down and change the eyes to blue and **bam!** Waffle Cat. Nothing a little image-editing can't do :)

IMO, that candy texture is ugly. The dog is OK I guess, but that candy texture is just bleh. The white background is just so out of place. When you said 'candy' I thought something like [this](http://vignette4.wikia.nocookie.net/cuttherope/images/1/1c/Candy.png/revision/latest?cb=20111028025349l) (but square).

**Comment by paramat** [comment]
The white seems ok to me because it is emitting maximum light level.

**Comment by DS-Minetest** [comment]
I think, I would more like something like the gnu than the pup. The pup doesn't seem to be that epic like the nyan cat and only for small kids => it doesn't have style. :sunglasses:

**Comment by paramat** [comment]
The choice of design will be very subjective and down to taste, i don't love this like i love the nyan cat but i'm trying to not be too fussy about what replaces it.

**Comment by paramat** [comment]
#1658 merged.

**Comment by VanessaE** [comment]
I guess this can be closed now.
